version: '3.7'

networks:
  network:
    driver: bridge

volumes:
  postgres-data:
    driver: local
  portainer-data:
    driver: local

services:

  todo:
    build:
      context: ../
      dockerfile: dockerX/todo/Dockerfile.local
      args:
         IMAGE: "python:3.7.1"
    restart: on-failure
    env_file:
      - todo/develop.env
    ports:
      - 8000:8000
    volumes:
      - ../todo:/usr/src/todo/
      - /opt/pycharm-2018.1.3/debug-eggs/:/usr/src/pycharm/debug-eggs/
    depends_on:
      - postgres
    networks:
      - network

  postgres:
    image: "postgres:11.1-alpine"
    ports:
      - 8998:5432
    volumes:
      - postgres-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=todo_user
      - POSTGRES_PASSWORD=wVlIie718fkg
      - POSTGRES_DB=todo_database
    networks:
      - network

  portainer:
    image: portainer/portainer
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - portainer-data:/data
    ports:
      - "8897:9000"
    networks:
      - network
